draft-ietf-quic-http.md

```diff
@@ -1102,69 +1167,33 @@ allowing time for any in-flight requests (at least one round-trip time), the
-graceful shutdown SHOULD use the QUIC APPLICATION_CLOSE frame with the
-HTTP_NO_ERROR code.
+Once all accepted requests have been processed, the server MAY permit the
+connection to become idle, or MAY initiate an explicit closure of the
+connection.  An endpoint that completes a graceful shutdown SHOULD use the
+HTTP_NO_ERROR code when closing the connection.
 
+## Explicit Application Closure
 
-### MAX_PUSH_ID {#frame-max-push-id}
-
-The MAX_PUSH_ID frame (type=0xD) is used by clients to control the number of
-server pushes that the server can initiate.  This sets the maximum value for a
-Push ID that the server can use in a PUSH_PROMISE frame.  Consequently, this
-also limits the number of push streams that the server can initiate in addition
-to the limit set by the QUIC MAX_STREAM_ID frame.
-
-The MAX_PUSH_ID frame is always sent on a control stream.  Receipt of a
-MAX_PUSH_ID frame on any other stream MUST be treated as a connection error of
-type HTTP_WRONG_STREAM.
-
-A server MUST NOT send a MAX_PUSH_ID frame.  A client MUST treat the receipt of
-a MAX_PUSH_ID frame as a connection error of type HTTP_MALFORMED_FRAME.
-
-The maximum Push ID is unset when a connection is created, meaning that a server
-cannot push until it receives a MAX_PUSH_ID frame.  A client that wishes to
-manage the number of promised server pushes can increase the maximum Push ID by
-sending a MAX_PUSH_ID frame as the server fulfills or cancels server pushes.
-
-~~~~~~~~~~  drawing
- 0                   1                   2                   3
- 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-|                          Push ID (i)                        ...
-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-~~~~~~~~~~
-{: #fig-max-push title="MAX_PUSH_ID frame payload"}
-
-The MAX_PUSH_ID frame carries a single variable-length integer that identifies
-the maximum value for a Push ID that the server can use (see
-{{frame-push-promise}}).  A MAX_PUSH_ID frame cannot reduce the maximum Push ID;
-receipt of a MAX_PUSH_ID that contains a smaller value than previously received
-MUST be treated as a connection error of type HTTP_MALFORMED_FRAME.
-
-A server MUST treat a MAX_PUSH_ID frame payload that does not contain a single
-variable-length integer as a connection error of type
-HTTP_MALFORMED_FRAME.
-
+An HTTP/QUIC implementation can explicitly close the QUIC connection at any
+time. This results in sending a QUIC APPLICATION_CLOSE frame to the peer; the
+error code in this frame indicates to the peer why the connection is being
+closed.  See {{errors}} for error codes which can be used when closing a
+connection.
 
-# Connection Management
+Before closing the connection, a GOAWAY MAY be sent to allow the client to retry
+requests.  Including the GOAWAY frame in the same packet as the QUIC
```

Comment:
"retry some requests" - the implication here is that GOAWAY is sent immediately before, which would limit its usefulness.